lib/rules/no-mixed-requires.js

```diff
@@ -11,6 +11,17 @@
 
 module.exports = function(context) {
 
+    var grouping = false,
+        allowCall = false,
+        options = context.options[0];
+
+    if (typeof options === "object") {
+        grouping = options.grouping;
+        allowCall = options.allowCall;
+    } else {
```

Comment:
Based on the documentation, it looks like you should have an object with two properties as a second element in the array. However, this else statement allows for pretty much anything as a second element in the array. I think this is unnecessary and might lead to confusion. I would just remove else statement all together.